x-pack/plugins/ai_infra/product_doc_base/server/services/search/perform_search.ts

```diff
@@ -5,29 +5,28 @@
  * 2.0.
  */
 
-import type { Client } from '@elastic/elasticsearch';
+import type { ElasticsearchClient } from '@kbn/core/server';
+import type { ProductDocumentationAttributes } from '@kbn/product-doc-common';
 
 // https://search-labs.elastic.co/search-labs/blog/elser-rag-search-for-relevance
 
-export const performSemanticSearch = async ({
+export const performSearch = async ({
   searchQuery,
+  size,
   index,
   client,
 }: {
   searchQuery: string;
-  index: string;
-  client: Client;
+  size: number;
+  index: string | string[];
+  client: ElasticsearchClient;
 }) => {
-  const results = await client.search({
+  const results = await client.search<ProductDocumentationAttributes>({
     index,
-    size: 3,
+    size,
```

Comment:
@pgayvallet I've noticed that we don't handle duplicate versions for product documentation here (i.e. by using `collapse`, _top hits_, etc.). I assume that the indices under `.kibana-ai-product-doc-*` will contain only the latest stack version, so it shouldn't be a concern. Can you confirm that?

Reply:
That's correct, we're only working with a single version of the documentation here.